crates/oxc_linter/src/rules/promise/no_return_wrap.rs

```diff
@@ -0,0 +1,541 @@
+use oxc_allocator::Box as OBox;
+use oxc_ast::{
+    AstKind,
+    ast::{ArrowFunctionExpression, CallExpression, Expression, FunctionBody, Statement},
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{CompactStr, Span};
+
+use crate::{AstNode, context::LintContext, rule::Rule, utils::is_promise};
+
+fn no_return_wrap_diagnostic(span: Span, issue: &ReturnWrapper) -> OxcDiagnostic {
+    let warn_msg: CompactStr = match issue {
```

Comment:
Using `CompactStr` is redundant, all these messages exceed the `CompactStr` inline size limit. using `&'static str` should be fine here?